simulation/behavior_tree_plugin/include/behavior_tree_plugin/vehicle/follow_trajectory_sequence/polyline_trajectory_follower.hpp

```diff
@@ -0,0 +1,101 @@
+#ifndef BEHAVIOR_TREE_PLUGIN__VEHICLE__TRAJECTORY_FOLLOWER_HPP_
+#define BEHAVIOR_TREE_PLUGIN__VEHICLE__TRAJECTORY_FOLLOWER_HPP_
+
+#include <behavior_tree_plugin/vehicle/vehicle_action_node.hpp>
+#include <optional>
+#include <scenario_simulator_exception/exception.hpp>
+#include <traffic_simulator/data_type/follow_trajectory.hpp>
+
+namespace entity_behavior
+{
+namespace vehicle
+{
+
+class PolylineTrajectoryFollower
+{
+public:
+  PolylineTrajectoryFollower() = default;
+  ~PolylineTrajectoryFollower() = default;
+
+  using PolylineTrajectoryParameter_ptr =
+    std::shared_ptr<traffic_simulator::follow_trajectory::Parameter<
+      traffic_simulator::follow_trajectory::Polyline>>;
+
+  PolylineTrajectoryParameter_ptr trajectory_parameter_m;
```

Comment:
nazwa parameter mi tutaj nie pasuje - to vektor z wierzcholkami ulegajacy modyfikacji, nazwalbym `trajectory_vec`, dotego.. ten typ mi tutaj tez nie pasuje.. czemu to jest w takim ns, moze tak:
```
  using Polyline_ptr =
    std::shared_ptr<traffic_simulator::follow_trajectory::Parameter<
      traffic_simulator::follow_trajectory::Polyline>>;
      Polyline_ptr trajectory_vec_m;
```